### DISCUSSION TITLE
Distorted visualization of EGSPHANT file, Windows 11

### DISCUSSION THREAD

**Discussion by Jake145** [nrc-cnrc/EGSnrc#1083] [Questions] [answered]
Hi everyone! I have an issue with visualizing egsphant file created from DICOM data reconstructed with ctcreate. On Victoria it comes out very distorted, however re-creating the phantom with other codes there is no issue, so it shouldn't be a problem with the DICOM data. It is odd because I have never had an issue with dicom images until now. The images are less than 512x512 for 200 slices, so there shouldn't be any issue with the macros files, correct? 

![names-vol-0](https://github.com/nrc-cnrc/EGSnrc/assets/55702455/3efc5bb9-37a7-4c8c-b77b-1604fd93015a)

**Comment by rtownson** [comment]
It might be worth double checking with dosxyz_show, if you have access to linux. Or open the .egsphant file in a text editor, does the image look distorted? Usually you can get an idea of the image from just looking at the text file. Does ctcreate give any warnings or errors? Post the output here. What did you use for a RAMP file?

**Reply by Jake145** [reply]
I tried opening it but I can't really get an idea much different from one that actually came out good (another DICOM set of images), ctcreate went smooth. In attachment the output and the phantom (all in .txt to send through github)
[OUTPUT.txt](https://github.com/nrc-cnrc/EGSnrc/files/13591866/OUTPUT.txt)
[PHANTOM.txt](https://github.com/nrc-cnrc/EGSnrc/files/13591867/PHANTOM.txt)

**Reply by rtownson** [reply, marked as answer]
I see that you used the default ramp. Does this actually match the appropriate range of CT values for your scanner?

It looks like the z range gets adjusted to "-4.8171   -1.7391" instead of "-4.8171   1.7391". Does that make sense for your image?

I'm entirely sure, but you may need to adjust the following lines in `HEN_HOUSE/omega/progs/ctcreate/ReadCT_dicom.c` and recompile it. As it suggests in the comments, you'll also need to adjust `CTIMAX` and `CTJMAX`. You could try setting these to 432.
```
/* MAX_CT_ROW_SIZE must exactly equal $CTIMAX in ctcreate.mortran. */
#define MAX_CT_ROW_SIZE 512

/* MAX_CT_COL_SIZE must exactly equal $CTJMAX in ctcreate.mortran. */
#define MAX_CT_COL_SIZE 512
```

**Reply by Jake145** [reply]
Your last suggestion (changing from 512 to 432) worked!! Thank you so much!
